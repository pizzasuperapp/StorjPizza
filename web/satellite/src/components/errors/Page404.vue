// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./page40X.html" />

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import LogoIcon from '@/../static/images/logo.svg';
import MainIcon from '@/../static/images/errors/404.svg';

// @vue/component
@Component({
    components: {
        LogoIcon,
        MainIcon,
    },
})
// 404 component on page not found error
export default class Page404 extends Vue {}
</script>

<style scoped lang="scss">
    .error-container {
        position: fixed;
        top: 0;
        left: 0;
        z-index: 1000;
        width: 100%;
        height: 100vh;
        background-color: #fff;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        -webkit-touch-callout: none;
        user-select: none;

        &__title {
            font-family: 'font_regular', sans-serif;
            margin-bottom: 60px;
            font-size: 32px;
        }
    }

    .logo {
        display: flex;
        align-items: center;
        position: absolute;
        top: 87px;
        left: 100px;
    }

    .text {
        margin-left: 15px;
    }
</style>
