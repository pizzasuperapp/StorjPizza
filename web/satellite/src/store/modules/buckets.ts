// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import {StoreModule} from "@/types/store";

export const BUCKET_ACTIONS = {
    FETCH: 'setBuckets',
    FETCH_ALL_BUCKET_NAMES: 'getAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    CLEAR: 'clearBuckets',
};

export const BUCKET_MUTATIONS = {
    SET: 'setBuckets',
    SET_ALL_BUCKET_NAMES: 'setAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    SET_PAGE: 'setBucketPage',
    CLEAR: 'clearBuckets',
};

const {
    FETCH,
    FETCH_ALL_BUCKET_NAMES,
} = BUCKET_ACTIONS;
const {
    SET,
    SET_ALL_BUCKET_NAMES,
    SET_PAGE,
    SET_SEARCH,
    CLEAR,
} = BUCKET_MUTATIONS;
const bucketPageLimit = 7;
const firstPage = 1;

export class BucketsState {
    public allBucketNames: string[] = [];
    public cursor: BucketCursor = { limit: bucketPageLimit, search: '', page: firstPage };
    public page: BucketPage = { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: bucketPageLimit, search: '', totalCount: 0 };
}

interface BucketsContext {
    state: BucketsState
    commit: (string, ...unknown) => void
    rootGetters: {
        selectedProject: {
            id: string
        }
    }
}

/**
 * creates buckets module with all dependencies
 *
 * @param api - buckets api
 */
export function makeBucketsModule(api: BucketsApi): StoreModule<BucketsState, BucketsContext> {
    return {
        state: new BucketsState(),

        mutations: {
            [SET](state: BucketsState, page: BucketPage) {
                state.page = page;
            },
            [SET_ALL_BUCKET_NAMES](state: BucketsState, allBucketNames: string[]) {
                state.allBucketNames = allBucketNames;
            },
            [SET_PAGE](state: BucketsState, page: number) {
                state.cursor.page = page;
            },
            [SET_SEARCH](state: BucketsState, search: string) {
                state.cursor.search = search;
            },
            [CLEAR](state: BucketsState) {
                state.allBucketNames = [];
                state.cursor = new BucketCursor('', bucketPageLimit, firstPage);
                state.page = new BucketPage([], '', bucketPageLimit, 0, 1, 1, 0);
            },
        },
        actions: {
            [FETCH]: async function({commit, rootGetters, state}: BucketsContext, page: number): Promise<BucketPage> {
                const projectID = rootGetters.selectedProject.id;
                const before = new Date();
                state.cursor.page = page;

                commit(SET_PAGE, page);

                const result: BucketPage = await api.get(projectID, before, state.cursor);

                commit(SET, result);

                return result;
            },
            [FETCH_ALL_BUCKET_NAMES]: async function({commit, rootGetters}: BucketsContext): Promise<string[]> {
                const result: string[] = await api.getAllBucketNames(rootGetters.selectedProject.id);

                commit(SET_ALL_BUCKET_NAMES, result);

                return result;
            },
            [BUCKET_ACTIONS.SET_SEARCH]: function({commit}: BucketsContext, search: string) {
                commit(SET_SEARCH, search);
            },
            [BUCKET_ACTIONS.CLEAR]: function({commit}: BucketsContext) {
                commit(CLEAR);
            },
        },
        getters: {
            page: (state: BucketsState): BucketPage => state.page,
            cursor: (state: BucketsState): BucketCursor => state.cursor,
        },
    };
}
