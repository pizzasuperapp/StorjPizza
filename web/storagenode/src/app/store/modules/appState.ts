// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const APPSTATE_MUTATIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART,',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK: 'SET_DARK',
    SET_NO_PAYOUT_INFO: 'SET_NO_PAYOUT_INFO',
    SET_LOADING_STATE: 'SET_LOADING_STATE',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

export const APPSTATE_ACTIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK_MODE: 'SET_DARK_MODE',
    SET_NO_PAYOUT_DATA: 'SET_NO_PAYOUT_DATA',
    SET_LOADING: 'SET_LOADING',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

const {
    TOGGLE_SATELLITE_SELECTION,
    TOGGLE_BANDWIDTH_CHART,
    TOGGLE_EGRESS_CHART,
    TOGGLE_INGRESS_CHART,
    CLOSE_ADDITIONAL_CHARTS,
    CLOSE_ALL_POPUPS,
    TOGGLE_PAYOUT_CALENDAR,
} = APPSTATE_MUTATIONS;

class AppState {
    public constructor (
        public isSatelliteSelectionShown = false,
        public isBandwidthChartShown = true,
        public isEgressChartShown = false,
        public isIngressChartShown = false,
        public isPayoutCalendarShown = false,
        public isDarkMode = false,
        public isNoPayoutData = false,
        public isLoading = true,
        public isPayoutHistoryCalendarShown = false,
    ) {}
}

interface AppContext {
    state: AppState;
    commit: (string, ...unknown) => void;
}

export const appStateModule = {
    state: new AppState(),
    mutations: {
        [TOGGLE_SATELLITE_SELECTION](state: AppState): void {
            state.isSatelliteSelectionShown = !state.isSatelliteSelectionShown;
        },
        [TOGGLE_BANDWIDTH_CHART](state: AppState): void {
            state.isBandwidthChartShown = !state.isBandwidthChartShown;
        },
        [TOGGLE_EGRESS_CHART](state: AppState): void {
            state.isEgressChartShown = !state.isEgressChartShown;
        },
        [TOGGLE_INGRESS_CHART](state: AppState): void {
            state.isIngressChartShown = !state.isIngressChartShown;
        },
        [TOGGLE_PAYOUT_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutCalendarShown = value;
        },
        [CLOSE_ADDITIONAL_CHARTS](state: AppState): void {
            state.isBandwidthChartShown = true;
            state.isIngressChartShown = false;
            state.isEgressChartShown = false;
        },
        [APPSTATE_MUTATIONS.SET_DARK](state: AppState, value: boolean): void {
            state.isDarkMode = value;
        },
        [APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO](state: AppState, value: boolean): void {
            state.isNoPayoutData = value;
        },
        [APPSTATE_MUTATIONS.SET_LOADING_STATE](state: AppState, value: boolean): void {
            state.isLoading = value;
        },
        [CLOSE_ALL_POPUPS](state: AppState): void {
            state.isSatelliteSelectionShown = false;
        },
        [APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutHistoryCalendarShown = value;
        },
    },
    actions: {
        [APPSTATE_ACTIONS.TOGGLE_SATELLITE_SELECTION]: function ({commit, state}: AppContext): void {
            if (!state.isSatelliteSelectionShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_SATELLITE_SELECTION);

                return;
            }

            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_CALENDAR, value);
        },
        [APPSTATE_ACTIONS.TOGGLE_EGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
        },
        [APPSTATE_ACTIONS.TOGGLE_INGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
        },
        [APPSTATE_ACTIONS.SET_DARK_MODE]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_DARK, value);
        },
        [APPSTATE_ACTIONS.SET_NO_PAYOUT_DATA]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO, value);
        },
        [APPSTATE_ACTIONS.SET_LOADING]: function ({commit}: AppContext, value: boolean): void {
            value ? commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value) :
                setTimeout(() => { commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value); }, 1000);
        },
        [APPSTATE_ACTIONS.CLOSE_ADDITIONAL_CHARTS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ADDITIONAL_CHARTS);
        },
        [APPSTATE_ACTIONS.CLOSE_ALL_POPUPS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR, value);
        },
    },
};
