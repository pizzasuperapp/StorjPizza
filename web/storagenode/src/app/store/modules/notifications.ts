// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/app/store';
import { NotificationsState, UINotification } from '@/app/types/notifications';
import { NotificationsService } from '@/storagenode/notifications/service';

export const NOTIFICATIONS_MUTATIONS = {
    SET_NOTIFICATIONS: 'SET_NOTIFICATIONS',
    MARK_AS_READ: 'MARK_AS_READ',
    READ_ALL: 'READ_ALL',
    SET_LATEST: 'SET_LATEST',
};

export const NOTIFICATIONS_ACTIONS = {
    GET_NOTIFICATIONS: 'getNotifications',
    MARK_AS_READ: 'markAsRead',
    READ_ALL: 'readAll',
};

interface NotificationsContext {
    state: NotificationsState;
    commit: (string, ...unknown) => void;
}

/**
 * creates notifications module with all dependencies
 *
 * @param service - payments service
 */
export function newNotificationsModule(service: NotificationsService): StoreModule<NotificationsState> {
    return {
        state: new NotificationsState(),
        mutations: {
            [NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS](state: NotificationsState, notificationsState: NotificationsState): void {
                state.notifications = notificationsState.notifications;
                state.pageCount = notificationsState.pageCount;
                state.unreadCount = notificationsState.unreadCount;
            },
            [NOTIFICATIONS_MUTATIONS.SET_LATEST](state: NotificationsState, notificationsState: NotificationsState): void {
                state.latestNotifications = notificationsState.notifications;
            },
            [NOTIFICATIONS_MUTATIONS.MARK_AS_READ](state: NotificationsState, id: string): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    if (notification.id === id) {
                        notification.markAsRead();
                    }

                    return notification;
                });
            },
            [NOTIFICATIONS_MUTATIONS.READ_ALL](state: NotificationsState): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    notification.markAsRead();

                    return notification;
                });

                state.unreadCount = 0;
            },
        },
        actions: {
            [NOTIFICATIONS_ACTIONS.GET_NOTIFICATIONS]: async function ({commit}: NotificationsContext, pageIndex: number): Promise<void> {
                const notificationsResponse = await service.notifications(pageIndex);

                const notifications = notificationsResponse.page.notifications.map(notification => new UINotification(notification));

                const notificationState = new NotificationsState(notifications, notificationsResponse.page.pageCount, notificationsResponse.unreadCount);

                commit(NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS, notificationState);

                if (pageIndex === 1) {
                    commit(NOTIFICATIONS_MUTATIONS.SET_LATEST, notificationState);
                }
            },
            [NOTIFICATIONS_ACTIONS.MARK_AS_READ]: async function ({commit}: NotificationsContext, id: string): Promise<void> {
                await service.readSingeNotification(id);
                commit(NOTIFICATIONS_MUTATIONS.MARK_AS_READ, id);
            },
            [NOTIFICATIONS_ACTIONS.READ_ALL]: async function ({commit}: NotificationsContext): Promise<void> {
                await service.readAllNotifications();

                commit(NOTIFICATIONS_MUTATIONS.READ_ALL);
            },
        },
    };
}
