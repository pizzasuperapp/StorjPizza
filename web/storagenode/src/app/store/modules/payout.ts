// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/app/store';
import {
    PayoutInfoRange,
    PayoutState,
} from '@/app/types/payout';
import { StorageNodeState } from '@/app/types/sno';
import { getHeldPercentage } from '@/app/utils/payout';
import { SizeBreakpoints } from '@/private/memory/size';
import {
    EstimatedPayout,
    PayoutPeriod,
    SatelliteHeldHistory,
    SatellitePayoutForPeriod,
    TotalPayments,
    TotalPaystubForPeriod,
} from '@/storagenode/payouts/payouts';
import { PayoutService } from '@/storagenode/payouts/service';

export const PAYOUT_MUTATIONS = {
    SET_PAYOUT_INFO: 'SET_PAYOUT_INFO',
    SET_RANGE: 'SET_RANGE',
    SET_TOTAL: 'SET_TOTAL',
    SET_HELD_PERCENT: 'SET_HELD_PERCENT',
    SET_HELD_HISTORY: 'SET_HELD_HISTORY',
    SET_ESTIMATION: 'SET_ESTIMATION',
    SET_PERIODS: 'SET_PERIODS',
    SET_PAYOUT_HISTORY: 'SET_PAYOUT_HISTORY',
    SET_PAYOUT_HISTORY_PERIOD: 'SET_PAYOUT_HISTORY_PERIOD',
    SET_PAYOUT_HISTORY_AVAILABLE_PERIODS: 'SET_PAYOUT_HISTORY_AVAILABLE_PERIODS',
};

export const PAYOUT_ACTIONS = {
    GET_PAYOUT_INFO: 'GET_PAYOUT_INFO',
    SET_PERIODS_RANGE: 'SET_PERIODS_RANGE',
    GET_TOTAL: 'GET_TOTAL',
    GET_HELD_HISTORY: 'GET_HELD_HISTORY',
    GET_ESTIMATION: 'GET_ESTIMATION',
    GET_PERIODS: 'GET_PERIODS',
    GET_PAYOUT_HISTORY: 'GET_PAYOUT_HISTORY',
    SET_PAYOUT_HISTORY_PERIOD: 'SET_PAYOUT_HISTORY_PERIOD',
};

// TODO: move to config in storagenode/payouts
export const BANDWIDTH_DOWNLOAD_PRICE_PER_TB = 2000;
export const BANDWIDTH_REPAIR_PRICE_PER_TB = 1000;
export const DISK_SPACE_PRICE_PER_TB = 150;

interface PayoutContext {
    rootState: {
        node: StorageNodeState;
    };
    state: PayoutState;
    commit: (string, ...unknown) => void;
}

/**
 * creates notifications module with all dependencies
 *
 * @param service - payments service
 */
export function newPayoutModule(service: PayoutService): StoreModule<PayoutState> {
    return {
        state: new PayoutState(),
        mutations: {
            [PAYOUT_MUTATIONS.SET_PAYOUT_INFO](state: PayoutState, totalPaystubForPeriod: TotalPaystubForPeriod): void {
                state.totalPaystubForPeriod = totalPaystubForPeriod;
            },
            [PAYOUT_MUTATIONS.SET_TOTAL](state: PayoutState, totalPayments: TotalPayments): void {
                state.totalPayments = totalPayments;
                state.currentMonthEarnings = totalPayments.currentMonthEarnings;
            },
            [PAYOUT_MUTATIONS.SET_RANGE](state: PayoutState, periodRange: PayoutInfoRange): void {
                state.periodRange = periodRange;
            },
            [PAYOUT_MUTATIONS.SET_HELD_PERCENT](state: PayoutState, heldPercentage: number): void {
                state.heldPercentage = heldPercentage;
            },
            [PAYOUT_MUTATIONS.SET_HELD_HISTORY](state: PayoutState, heldHistory: SatelliteHeldHistory[]): void {
                state.heldHistory = heldHistory;
            },
            [PAYOUT_MUTATIONS.SET_ESTIMATION](state: PayoutState, estimatedInfo: EstimatedPayout): void {
                state.estimation = estimatedInfo;
            },
            [PAYOUT_MUTATIONS.SET_PERIODS](state: PayoutState, periods: PayoutPeriod[]): void {
                state.payoutPeriods = periods;
            },
            [PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY](state: PayoutState, payoutHistory: SatellitePayoutForPeriod[]): void {
                state.payoutHistory = payoutHistory;
            },
            [PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY_PERIOD](state: PayoutState, period: string): void {
                state.payoutHistoryPeriod = period;
            },
            [PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY_AVAILABLE_PERIODS](state: PayoutState, periods: PayoutPeriod[]): void {
                state.payoutHistoryAvailablePeriods = periods;
            },
        },
        actions: {
            [PAYOUT_ACTIONS.GET_PAYOUT_INFO]: async function ({ commit, state, rootState }: PayoutContext, satelliteId = ''): Promise<void> {
                const totalPaystubForPeriod = await service.paystubSummaryForPeriod(
                    state.periodRange.start,
                    state.periodRange.end,
                    satelliteId,
                );

                commit(PAYOUT_MUTATIONS.SET_HELD_PERCENT, getHeldPercentage(rootState.node.selectedSatellite.joinDate));
                commit(PAYOUT_MUTATIONS.SET_PAYOUT_INFO, totalPaystubForPeriod);
            },
            [PAYOUT_ACTIONS.GET_TOTAL]: async function ({ commit, rootState }: PayoutContext, satelliteId = ''): Promise<void> {
                const now = new Date();
                const start = new PayoutPeriod(rootState.node.selectedSatellite.joinDate.getUTCFullYear(), rootState.node.selectedSatellite.joinDate.getUTCMonth());
                const end = new PayoutPeriod(now.getUTCFullYear(), now.getUTCMonth());

                const totalPayments = await service.totalPayments(start, end, satelliteId);

                // TODO: move to service
                const currentBandwidthDownload = (rootState.node.egressChartData || [])
                    .map(data => data.egress.usage)
                    .reduce((previous, current) => previous + current, 0);

                const currentBandwidthAuditAndRepair = (rootState.node.egressChartData || [])
                    .map(data => data.egress.audit + data.egress.repair)
                    .reduce((previous, current) => previous + current, 0);

                const approxHourInMonth = 720;
                const currentDiskSpace = (rootState.node.storageChartData || [])
                    .map(data => data.atRestTotal)
                    .reduce((previous, current) => previous + current, 0) / approxHourInMonth;

                const thisMonthEarnings = (currentBandwidthDownload * BANDWIDTH_DOWNLOAD_PRICE_PER_TB
                    + currentBandwidthAuditAndRepair * BANDWIDTH_REPAIR_PRICE_PER_TB
                    + currentDiskSpace * DISK_SPACE_PRICE_PER_TB) / SizeBreakpoints.TB;

                totalPayments.setCurrentMonthEarnings(thisMonthEarnings);

                commit(PAYOUT_MUTATIONS.SET_HELD_PERCENT, getHeldPercentage(rootState.node.selectedSatellite.joinDate));
                commit(PAYOUT_MUTATIONS.SET_TOTAL, totalPayments);
            },
            [PAYOUT_ACTIONS.SET_PERIODS_RANGE]: function ({ commit }: PayoutContext, periodRange: PayoutInfoRange): void {
                commit(PAYOUT_MUTATIONS.SET_RANGE, periodRange);
            },
            [PAYOUT_ACTIONS.GET_HELD_HISTORY]: async function ({ commit }: PayoutContext): Promise<void> {
                const heldHistory = await service.allSatellitesHeldHistory();

                commit(PAYOUT_MUTATIONS.SET_HELD_HISTORY, heldHistory);
            },
            [PAYOUT_ACTIONS.GET_PERIODS]: async function ({commit}: PayoutContext, satelliteId = ''): Promise<void> {
                const periods = await service.availablePeriods(satelliteId);

                commit(PAYOUT_MUTATIONS.SET_PERIODS, periods);

                if (!satelliteId) {
                    commit(PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY_AVAILABLE_PERIODS, periods);
                }
            },
            [PAYOUT_ACTIONS.GET_ESTIMATION]: async function ({ commit }: PayoutContext, satelliteId = ''): Promise<void> {
                const estimatedInfo = await service.estimatedPayout(satelliteId);

                commit(PAYOUT_MUTATIONS.SET_ESTIMATION, estimatedInfo);
            },
            [PAYOUT_ACTIONS.GET_PAYOUT_HISTORY]: async function ({ commit, state }: PayoutContext): Promise<void> {
                if (!state.payoutHistoryPeriod) return;

                const payoutHistory = await service.payoutHistory(state.payoutHistoryPeriod);

                commit(PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY, payoutHistory);
            },
            [PAYOUT_ACTIONS.SET_PAYOUT_HISTORY_PERIOD]: function ({ commit }: PayoutContext, period: string): void {
                commit(PAYOUT_MUTATIONS.SET_PAYOUT_HISTORY_PERIOD, period);
            },
        },
        getters: {
            totalPaidForPayoutHistoryPeriod: (state: PayoutState): number => {
                return state.payoutHistory.map(data => data.paid)
                    .reduce((previous, current) => previous + current, 0);
            },
        },
    };
}
