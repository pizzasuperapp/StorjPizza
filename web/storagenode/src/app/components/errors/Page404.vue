// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./page40X.html" />

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import { APPSTATE_ACTIONS } from '@/app/store/modules/appState';

// @vue/component
@Component
// 404 component on page not found error
export default class Page404 extends Vue {
    public beforeMount(): void {
        this.$store.dispatch(APPSTATE_ACTIONS.SET_LOADING, false);
    }
}
</script>

<style scoped lang="scss">
    .error-container {
        position: fixed;
        top: 0;
        left: 0;
        z-index: 1000;
        width: 100%;
        height: 100vh;
        background-color: var(--error-page-background-color);
        color: var(--error-page-text-color);
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        -webkit-touch-callout: none;
        user-select: none;
        text-align: center;

        &__main-image {
            width: 320px;
            height: 311px;
            background: var(--error-page-image-path);
            background-size: contain;
        }

        &__title {
            margin-block-start: 0;
            margin-block-end: 0;
            font-family: 'font_bold', sans-serif;
            margin-top: 46px;
            font-size: 42px;
            line-height: 42px;
        }

        &__text {
            margin-block-start: 0;
            margin-block-end: 0;
            font-family: 'font_regular', sans-serif;
            margin: 25px 0 36px;
            font-size: 16px;
            line-height: 16px;
        }

        &__button {
            padding: 16px 24px;
            background: var(--error-page-button-background);
            border-radius: 6px;
            color: white;
            font-family: 'font_bold', sans-serif;
            font-size: 14px;
            line-height: 14px;
            text-decoration: none;
        }
    }

    @media screen and (max-width: 630px) {

        .error-container {
            padding: 0 27px;
            width: calc(100% - 54px);

            &__main-image {
                margin-top: 89px;
                width: 184px;
                height: 179px;
            }

            &__title {
                margin-top: 27px;
                font-size: 32px;
                line-height: 44px;
            }

            &__text {
                margin: 12px 0 20px;
                line-height: 24px;
            }
        }
    }
</style>
