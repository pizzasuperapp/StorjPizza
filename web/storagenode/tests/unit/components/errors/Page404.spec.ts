// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import Page404 from '@/app/components/errors/Page404.vue';

import { appStateModule } from '@/app/store/modules/appState';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({ modules: { appStateModule }});

describe('Page404', (): void => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(Page404, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
