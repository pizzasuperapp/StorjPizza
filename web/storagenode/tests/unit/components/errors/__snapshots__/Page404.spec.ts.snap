// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 renders correctly 1`] = `
<div class="error-container">
  <div class="error-container__main-image"></div>
  <h1 class="error-container__title">404. Something Went Wrong</h1>
  <h3 class="error-container__text">The page you’re trying to access is either broken, or doesn’t exist</h3> <a href="/" class="error-container__button">Back to home</a>
</div>
`;
